lib/src/test/java/io/ably/lib/test/rest/HttpHeaderTest.java

```diff
@@ -83,12 +83,50 @@ public void header_lib_channel_publish() {
             Assert.assertNotNull("Expected headers", headers);
             Assert.assertEquals(headers.get("x-ably-version"), "2");
             Assert.assertEquals(headers.get("ably-agent"), expectedAblyAgentHeader);
+            // RSA7e2
+            Assert.assertNull("Shouldn't include 'x-ably-clientid' if `clientId` is not specified", headers.get("x-ably-clientid"));
         } catch (AblyException e) {
             e.printStackTrace();
             Assert.fail("header_lib_channel_publish: Unexpected exception");
         }
     }
 
+    /**
+     * The header `X-Ably-ClientId`
+     * should be included in all REST requests to the Ably endpoint
+     * if {@link ClientOptions#clientId} is specified
+     * <p>
+     * Spec: RSA7e2
+     * </p>
+     */
+    @Test
+    public void header_client_id_on_channel_publish() {
+        try {
+            /* Init values for local server */
+            ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+            opts.environment = null;
+            opts.tls = false;
+            opts.port = server.getListeningPort();
+            opts.restHost = "localhost";
+            opts.clientId = "test client";
+            AblyRest ably = new AblyRest(opts);
+
+            /* Publish message */
+            String messageName = "test message";
+            String messageData = String.valueOf(System.currentTimeMillis());
+
+            Channel channel = ably.channels.get("test");
+            channel.publish(messageName, messageData);
+
+            /* Get last headers */
+            Map<String, String> headers = server.getHeaders();
+            Assert.assertEquals(headers.get("x-ably-clientid"), "dGVzdCBjbGllbnQ=");
```

Comment:
```suggestion
            Assert.assertEquals(headers.get("x-ably-clientid"),Base64Coder.encodeString("test client"));
```

Reply:
I prefer not to put logic for expected value. Potential bug in `Base64Coder#encodeString` method can make test unreliable

Reply:
I'll add comment that it is in fact base64 encoded "test client" value, to make it more readable

Reply:
Sure, no issues. Just that, the assertion was more readable. Agree that assertions should be dumb and shouldn't be any logic into them.

Reply:
lg